synapse/lib/layer.py

```diff
@@ -4094,91 +4094,114 @@ async def iterNodeDataKeys(self, buid):
             prop = self.getAbrvProp(abrv)
             yield prop[0]
 
-    async def iterLayerAddPerms(self):
+    async def confirmLayerEditPerms(self, user, gateiden=None, delete=False):
+        if gateiden is None:
+            gateiden = self.iden
```

Comment:
def a nit, but is there a use case for checking perms on the self layer? perhaps gateiden should be required?

Reply:
Yeah, the delete case here uses the self layer: https://github.com/vertexproject/synapse/pull/3750/files#diff-cb95e16773a89fbc2b7cbc1048f93c9ab9f26cc7b2a41bf31380c9892a44381cR1457

Reply:
I see your point though. I can make gateiden required.